services/workflows-service/src/business-report/business-report.service.ts

```diff
@@ -20,10 +22,68 @@ export class BusinessReportService {
     return await this.businessReportRepository.findMany(args, projectIds);
   }
 
-  async findFirst<T extends Prisma.BusinessReportFindFirstArgs>(
+  async upsert<T extends Prisma.BusinessReportUpsertArgs>(
+    args: Prisma.SelectSubset<T, Prisma.BusinessReportUpsertArgs>,
+    projectIds: NonNullable<TProjectIds>,
+  ) {
+    if (!args.where.id) {
+      return await this.businessReportRepository.create({ data: args.create });
+    }
+
+    await this.businessReportRepository.updateMany({
+      where: {
+        id: args.where.id,
+        project: { id: { in: projectIds } },
+      },
+      data: args.update,
+    });
+
+    return await this.businessReportRepository.findFirstOrThrow(
+      {
+        where: {
+          id: args.where.id,
+        },
+      },
+      projectIds,
+    );
+  }
```

Comment:
I'd expect Prisma's `upsert` to be used here. Consider at least changing the name.